Chessington.GameEngine/Pieces/King.cs

```diff
@@ -10,7 +10,28 @@ public King(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            var initialRow = board.FindPiece(this).Row;
+            var initialCol = board.FindPiece(this).Col;
+            var availableMoves = new List<Square>();
+            
+            for (int i = -1; i <= 1; i++)
+            {
+                for (int j = -1; j <= 1; j++)
+                {
+                    if (CheckPosition(initialRow + i, initialCol + j, board))
+                    {
+                        availableMoves.Add(Square.At(initialRow + i, initialCol + j));
+                    }
+                }
+            }
+            
+            availableMoves.RemoveAll(x => x == Square.At(initialRow,initialCol));
+            
+            return availableMoves;
+        }
+        private bool CheckPosition(int row, int col, Board board)
```

Comment:
check position is repeated in most of the classes you can put it on the board class and reuse it?